src/examples/zkapps/dex/dex-with-actions.ts

```diff
@@ -0,0 +1,367 @@
+/**
+ * This DEX implementation differs from ./dex.ts in two ways:
+ * - More minimal & realistic; stuff only designed for testing protocol features was removed
+ * - Uses an "async" pattern with actions that lets users claim funds later & reduces account updates
```

Comment:
```suggestion
 * - Uses an async pattern with actions that lets users claim funds later and reduces account updates
```

not sure we enclose "async" with double quotation marks. Is this an API convention? we can remove